### DISCUSSION TITLE
[go API] `instance.OrphanedFiles` ignored at build

### DISCUSSION THREAD

**Discussion by b4nst** [cue-lang/cue#2452] [Q&A] [answered]
Hey there!
When using `load.Instances` with a mix of files, the api puts the YAML ones into `instance.OrphanedFiles` (when `load.Config.DataFiles = true`.
It seems those files are ignored when later build with `context.BuildInstance(i)`. Is there a way to include them in the build process?

**Comment by verdverm** [comment, marked as answer]
Yup, experienced this as well. We manually process them

- (v1) https://github.com/hofstadter-io/hof/blob/_dev/lib/cuetils/runtime.go
- (v2) https://github.com/hofstadter-io/hof/blob/_dev/lib/runtime/load.go      (derived from v1, this is where we will continue making improvements and enhancements)

Some things to note,

- You have to add the syntax for the data files to the build instance before building
- You have to build up enough AST that CUE will be happy, data format specific
- We are using `data.json@path.in.value` rather than the `-l` flag, we plan to add this support later on, it's a bit more work because you may need to evaluate the `-l` flag value as CUE with the data as context, not sure if there is a chicken/egg issue here... tbd
- there are probably shenanigans on how the package is assigned to the data, need to think through this as well, because different data could ideally land in different packages
- our data placement uses `@` to place the full data somewhere, and is on a per file basis, whereas the cue `-l` flag works across all data files. Supporting both modes seems ideal for covering the different usage patterns. It is unclear where the overlap is for the `@embed()`. (i.e. I want a file included with my CUE module vs I expect the user to provide this file when using my module)

**Reply by myitcv** [reply]
See also https://github.com/cue-lang/cue/issues/1410
